servers/rendering/renderer_rd/storage_rd/texture_storage.cpp

```diff
@@ -643,6 +643,97 @@ void TextureStorage::canvas_texture_set_texture_repeat(RID p_canvas_texture, RS:
 	ct->clear_sets();
 }
 
+TextureStorage::CanvasTextureInfo TextureStorage::canvas_texture_get_info(RID p_texture, RS::CanvasItemTextureFilter p_base_filter, RS::CanvasItemTextureRepeat p_base_repeat, bool p_use_srgb, bool p_texture_is_data) {
```

Comment:
This is a copy of `canvas_texture_get_uniform_set`, but does not create a uniform set, and also provides the cached RIDs and state, to be used by the batch rendering. 

The intent is that a future iteration of the canvas batching could combine multiple textures, samplers, etc into an array, further reducing draw calls. Rendering large blocks of text would benefit from this optimisation, as they switch between textures for the same `TYPE_RECT`, resulting in many individual draw calls.